src/v/storage/parser.cc

```diff
@@ -93,15 +93,19 @@ static ss::future<result<iobuf>> verify_read_iobuf(
 }
 
 ss::future<result<stop_parser>> continuous_batch_parser::consume_header() {
-    return read_iobuf_exactly(_input, model::packed_record_batch_header_size)
+    auto f = ss::make_ready_future<result<model::record_batch_header>>(_header);
+    if (_header.header_crc == 0 && _header.size_bytes == 0) {
+      f = read_iobuf_exactly(_input, model::packed_record_batch_header_size)
       .then([this](iobuf b) -> result<iobuf> {
           if (b.empty()) {
               // benign outcome. happens at end of file
               return parser_errc::end_of_stream;
           }
-          if (b.size_bytes() != model::packed_record_batch_header_size) {
+          if (
```

Comment:
these next couple of lines look like a bad formatting? what changed.

Reply:
I've split changes from formatting, formatting is applied in next commit.